wordpress-mobile/WordPress-Android
# [Compile Warnings As Errors] WordPress Module - Resolve Deprecated Warnings - Part.2

Parent: #17173
Partially Closes: #17181

This PR resolves one part of all `Deprecated` related warnings for the `WordPress` module, and in this part, the most generic ones and specific to the main, debug, wordpress and jetpack sources:

Main Source (`247` Leftovers + `118`):
- About `247` warnings x `'XYX' is deprecated. Deprecated in Java` (Leftovers from [part.1 PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17246))
- `5` warnings x `'Xyz' is deprecated. Overrides deprecated member in 'xyz'. Deprecated in Java`
- `24` warnings x `'xyz' is deprecated. Use xyz instead.`
- `2` warnings x `'Xyz' is deprecated. This should be removed once we start receiving Xyz the source block from the backend`
- `87` warnings x `'Xyz' is deprecated. This class is being refactored, if you implement any change, please also update xyz`

Debug/WordPress/Jetpack Source (`11`):
- `11` warnings x `'Xyz' is deprecated. Deprecated in Java`

-----

Warnings Resolution List:

1. [Resolve kotlin stdlib's to upper case deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/d42665649fcaf57002bfd30fa6c0efe56bedb872)
2. [Resolve kotlin stdlib's to lower case deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/d2c76371534179f8f578f6e526d9114c1c04c5ca)
3. [Resolve kotlin stdlib's sum by deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/edc5be360486f76085fe6e80ae7a2350b353cb48)

Warnings Suppression List:

1. [Suppress on activity for result deprecated warnings for java.](https://github.com/wordpress-mobile/WordPress-Android/commit/f42ff80236ee4dfe952c30c3a6d4e084e9d410c4) (Associated Issue: #17249)
2. [Suppress on activity for result override deprecation warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/65fc483dd64470cc13133175b8891743d99573d9)
3. [Suppress avatar deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/8618a7a39696af55f9c091b84e8489d34ed0551e)
4. [Suppress kotlin stdlib's capitalize deprecated warning.](https://github.com/wordpress-mobile/WordPress-Android/commit/4ee513bf65a99d48ff6269d76f76471f0a8efee3)
5. [Suppress all photo picker related deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/99d1e8fc69ac29efffe750aa2f9ef38194a7651a)
6. [Suppress on request perm. result override deprecation warns.](https://github.com/wordpress-mobile/WordPress-Android/commit/ed119fa8aa4dca1314b89e09927b616e412f860e)
7. [Suppress google app client & fb instance id deprecated warns.](https://github.com/wordpress-mobile/WordPress-Android/commit/a58a9a47fb1f639c0b3610705015046c3144887d) (Associated Issue: #17229)
8. [Suppress connectivity manager & network info deprecated warns.](https://github.com/wordpress-mobile/WordPress-Android/commit/3432ea1e6eec89840e921ed215b1516010524a9c) (Associated Issue: #17230)
9. [Suppress toast deprecated warnings for debug source.](https://github.com/wordpress-mobile/WordPress-Android/commit/fd36cfafcf96ed27abe30bb76f767b78a5056985)
10. [Suppress system ui deprecated warnings for wordpress source.](https://github.com/wordpress-mobile/WordPress-Android/commit/d3bd7f51654cec61934fb9ff29387a0e601f2532) (Associated Issue: #17256)
11. [Suppress on activity created override deprecation warns.](https://github.com/wordpress-mobile/WordPress-Android/commit/2f85e7167f58da88d0a930a7f1f132ab13a418fb)
12. [Suppress wp activity utils deprecation warns for jetpack src.](https://github.com/wordpress-mobile/WordPress-Android/commit/ceae5bcbabc6d2d3debc64ba2d906530e54ba39f)

In addition to the above warning resolution list, the below `Detekt` warnings got resolved as well, leaving Detekt's `baseline.xml` file 2 entries shorter:

1. [Resolve detekt warnings by manually updating the baseline.](https://github.com/wordpress-mobile/WordPress-Android/commit/7eb56077bb38b72573bb2eaadfd77822447db382)

-----

PS: @irfano (plus @ovitrif) I added you as the main reviewer(s), that is, in addition to @wordpress-mobile/apps-infrastructure team itself, but randomly, since I want someone from the WordPress Android team to primarily sign-off on that change. 🥇

FYI: I am going to randomly add more of you in those PRs that will follow, just so you become more aware of this change and how close we are on enabling allWarningsAsErrors by default everywhere. 🎉

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could quickly smoke test both, the `WordPress` and `Jetpack` apps, and see if they are both working as expected.

-----

## Regression Notes

1. Potential unintended areas of impact

N/A

14. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

15. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
